benchmarking/tpcds/__main__.py

```diff
@@ -34,25 +54,84 @@ class RunArgs:
     query_indices: list[int]
     ray_address: Optional[str]
     dry_run: bool
+    validate: bool
+    print_schema: bool
 
 
 @dataclass
 class Result:
     index: int
     duration: Optional[timedelta]
     error_msg: Optional[str]
+    is_correct: Optional[bool] = None
+    validation_error: Optional[str] = None
 
     def __repr__(self) -> str:
         if self.duration and self.error_msg:
             typing.assert_never("Both duration and error_msg are not None")
         elif self.duration:
-            return f"(Q{self.index} SUCCESS - duration: {self.duration})"
+            validation_str = ""
+            if self.is_correct is not None:
+                validation_str = f" - Correct: {self.is_correct}"
+                if not self.is_correct and self.validation_error:
+                    validation_str += f" - Validation Error: {self.validation_error}"
+            return f"(Q{self.index} SUCCESS - duration: {self.duration}{validation_str})"
         elif self.error_msg:
             return f"(Q{self.index} FAILURE - error msg: {self.error_msg})"
         else:
             typing.assert_never("Both duration and error_msg are None")
 
 
+def setup_duckdb_catalog(data_dir: Path) -> duckdb.DuckDBPyConnection:
+    """Setup DuckDB connection with the same data files."""
+    conn = duckdb.connect(database=":memory:")
```

Comment:
the datagen generates a duckdb `.db` file in the data_dir. 

`duckdb.connect(f"{data_dir}/tpcds.db")`